core/src/main/java/com/kin/ecosystem/core/network/ApiClient.java

```diff
@@ -576,7 +576,7 @@ private Error deserializeError(String respBody) {
             return (Error) (json.deserialize(respBody, new TypeToken<Error>() {
             }.getType()));
         } catch (Throwable throwable) {
-            return null;
+            return new Error("Could not deserialize Error", "Could not deserialize: " + respBody, 0);
```

Comment:
Consider adding a second parameter i.e deserializeError(String respBody, String origin)  the origin will be used to track where deserializeError called from

Reply:
not sure why do we need it.
it can be called only from the client.